htdocs/compta/bank/class/api_bankaccounts.class.php

```diff
@@ -170,6 +170,8 @@ public function post($request_data = null)
 		}
 		// Date of the initial balance (required to create an account).
 		$account->date_solde = time();
+		// Use the value of date_solde sent through API otherwise let date_solde value to now which was initialised just before
```

Comment:
Can you test this other fix:  Just move the line
$account->date_solde = time();
at line 162 before the foreach ?

Reply:
Hello Laurent,

Indeed, that worked as you said

```
                // Date of the initial balance (required to create an account).
                $account->date_solde = time();
                foreach ($request_data as $field => $value) {
                        if ($field === 'caller') {
                                // Add a mention of caller so on trigger called after action, we can filter to avoid a loop if we try to sync back again with the caller
                                $account->context['caller'] = sanitizeVal($request_data['caller'], 'aZ09');
                                continue;
                        }

                        $account->$field = $this->_checkValForAPI($field, $value, $account);
                }
```

Reply:
Yes, you must get your branch, edit code to follow the change suggested (so just moving the line with no need to add the 
if (isset($request_data['date_solde'])) { ...
And push. This should update the PR automatically if you take the same branch than the one you used for your first push...

Reply:
Hello Laurent,
Thank you for your message.
I think I managed to do the change.

Reply:
I don't think so. PR still has old code. Did you repush after the change ?
You can also cancel pr and restart it from scratch as it is just 1 line move ...